src/daft-logical-plan/src/ops/source.rs

```diff
@@ -13,16 +18,61 @@ pub struct Source {
 
     /// Information about the source data location.
     pub source_info: Arc<SourceInfo>,
+    pub stats_state: StatsState,
 }
 
 impl Source {
     pub fn new(output_schema: SchemaRef, source_info: Arc<SourceInfo>) -> Self {
         Self {
             output_schema,
             source_info,
+            stats_state: StatsState::NotMaterialized,
         }
     }
 
+    pub(crate) fn with_materialized_scan_source(
```

Comment:
`with_`  is a bit misleading here. What about naming this 

`to_materialized_scan_source` or `build_materialized_scan_source `